c2corg_ui/static/js/documentediting.js

```diff
@@ -595,6 +594,46 @@ app.DocumentEditingController.prototype.updateMaxSteps = function(waypointType)
 
 
 /**
+ * Check what properties are missing and tell the user.
+ * @param {Object} doc
+ * @export
+ */
+app.DocumentEditingController.prototype.checkMissingProps = function(doc) {
+  var type = doc.type ? app.utils.getDoctype(doc.type) : this.modelName_ + 's'; // missing 's' in modelName if creating a new document
+  var fields = app.constants.REQUIRED_FIELDS[type];
+  // understandable structure by alertService
+  var missing = {'data': {'errors': []}};
+  var error;
+  for (var i = 0; i < fields.length; i++) {
+    var field = fields[i];
+    error = missing['data']['errors'][i] = {};
```

Comment:
I don't understand this code.
Doesn't this mean that a "missing" error is added for all required fields?
Shouldn't you add the error only after checking that the field is not set?